tests/unit/test_ask_tell_optimization.py

```diff
@@ -427,3 +436,74 @@ def __deepcopy__(self, memo: dict[int, object]) -> _UncopyableModel:
     with pytest.raises(NotImplementedError):
         ask_tell.to_result()
     assert ask_tell.to_result(copy=False).final_result.is_ok
+
+
+# Check that the correct dataset is routed to the model.
+# Note: this test is almost identical to the one in test_bayesian_optimizer.py.
+@pytest.mark.parametrize("use_global_model", [True, False])
+@pytest.mark.parametrize("use_global_init_dataset", [True, False])
+@pytest.mark.parametrize("num_query_points_per_batch", [1, 2])
+def test_ask_tell_optimizer_creates_correct_datasets_for_rank3_points(
+    use_global_model: bool, use_global_init_dataset: bool, num_query_points_per_batch: int
+) -> None:
+    batch_size = 4
+    if use_global_init_dataset:
+        init_data = {OBJECTIVE: mk_dataset([[0.5], [1.5]], [[0.25], [0.35]])}
+    else:
+        init_data = {
+            LocalizedTag(OBJECTIVE, i): mk_dataset([[0.5 + i], [1.5 + i]], [[0.25], [0.35]])
+            for i in range(batch_size)
+        }
+        init_data[OBJECTIVE] = mk_dataset([[0.5], [1.5]], [[0.25], [0.35]])
+
+    query_points = tf.reshape(
+        tf.constant(range(batch_size * num_query_points_per_batch), tf.float64),
+        (num_query_points_per_batch, batch_size, 1),
+    )
+
+    class DatasetChecker(QuadraticMeanAndRBFKernel, PseudoTrainableProbModel):
```

Comment:
Can you define this at the module level and import it from test_baysian_optimizer?